### DISCUSSION TITLE
Caching strategy for Cloudflare Workers/Redis/...?

### DISCUSSION THREAD

**Discussion by SebJansen** [nuxt/nuxt#16163] [Questions]
I want to implement caching middleware, to work on Cloudflare Workers, because they are quirky and they don't support CDN-ing with cache control headers. Workers run in front of the CDN, so headers have no impact. Instead, fetching from and writing to Edge cache happens through the Cache API.

Currently, I'm lost, but I think I need to do the following:
1. Run a specific middleware first [fetch a cached response), before all other middlewares
2. Run another specific middleware last [save a cached response], after all other middlewares

Does anybody have ideas, especially concerning point #2? 
I'm assuming that I need to use server middleware for point #1, and the lifecycle event app:rendered for point #2

Regarding point #2, this is as far as I made it:
```js
import {defineNuxtPlugin} from "#app";

export default defineNuxtPlugin(nuxtApp => {
    nuxtApp.hook('app:rendered', () => { console.log(nuxtApp.ssrContext.res) })
   // How to access raw HTML? 
})
```

Maybe there's some sort of next() that I can call in the server middleware, to get the SSR'd response? 

This concern also applies to other ways of caching, such as to and from Redis

**Comment by Plinpod** [comment]
Hey! Running into the same scenario. Did you ever find a solution for this?
